pkg/apis/eksctl.io/v1alpha5/validation.go

```diff
@@ -147,6 +147,23 @@ func ValidateClusterConfig(cfg *ClusterConfig) error {
 		return errors.New("vpc.manageSharedNodeSecurityGroupRules must be enabled when using ekstcl-managed security groups")
 	}
 
+	if err := validateKarpenterConfig(cfg); err != nil {
+		return fmt.Errorf("failed to validate karpenter config: %w", err)
+	}
+
+	return nil
+}
+
+func validateKarpenterConfig(cfg *ClusterConfig) error {
+	if cfg.Karpenter == nil {
+		return nil
+	}
+	if cfg.Karpenter.Version == "" {
+		return errors.New("version field is required if installing Karpenter is enabled")
+	}
+	if len(cfg.FargateProfiles) > 0 && IsDisabled(cfg.IAM.WithOIDC) {
+		return errors.New("iam.withOIDC must be enabled with Karpenter and Fargate profiles")
+	}
```

Comment:
isn't oidc just always required? Since we create the IRSA?

Reply:
Since the default behaviour for Karpenter is to create service accounts for IRSA, we should disallow setting `withOIDC` to false unless `karpenter.CreateServiceAccount` is explicitly set to `true`.

Reply:
We had a chat about this, and what we'll do is, require oidc always, and in case createServiceAccount is set to true, we pass in the created Role ARN to the helm chart as a service account annotation defined by spec here: https://github.com/aws/karpenter/blob/main/charts/karpenter/values.yaml#L7

Reply:
It just makes our lives easier.

Reply:
Makes sense, but we don't seem to have validation that disallows disabling `iam.withOIDC` when Karpenter is enabled. 

eksctl will still allow this config:

```yaml
apiVersion: eksctl.io/v1alpha5
kind: ClusterConfig

metadata:
  name: cluster-with-karpenter
  region: us-west-2
  version: '1.20'

karpenter:
  version: '0.4.3'

iam:
  withOIDC: false
```

Reply:
It's not yet implemented, correct. :)

Reply:
that is what I'm doing right now, sorry, that one might have not been clear